src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java

```diff
@@ -98,9 +141,37 @@ private void bindDataToTableView() {
         estimatedDeliveryDateColumn.setCellValueFactory(new PropertyValueFactory<>("estimatedDeliveryDate"));
         deliveryDateColumn.setCellValueFactory(new PropertyValueFactory<>("deliveryDate"));
 
-        // Set the items in the table
         tableView.getItems().setAll(clientOrders);
+    }
+
+    private void setEditEvents() {
+        clientIdColumn.setOnEditCommit(event -> {
+            ClientOrder order = event.getRowValue();
+            order.setClientId(event.getNewValue());
+            updateInDatabase(order);
+        });
+
+        productIdColumn.setOnEditCommit(event -> {
+            ClientOrder order = event.getRowValue();
+            order.setProductId(event.getNewValue());
+            updateInDatabase(order);
+        });
+
+        quantityColumn.setOnEditCommit(event -> {
+            ClientOrder order = event.getRowValue();
+            order.setQuantity(event.getNewValue());
+            updateInDatabase(order);
+        });
+
+        statusColumn.setOnEditCommit(event -> {
+            ClientOrder order = event.getRowValue();
+            order.setStatus(event.getNewValue());
+            updateInDatabase(order);
+        });
+    }
 
+    private void updateInDatabase(ClientOrder order) {
```

Comment:
Make the update service method and use it here to update. I'll add a Toast system later for confirming successful operation/signaling error.